src/rust/lib/graph-editor/src/component/edge.rs

```diff
@@ -500,9 +749,60 @@ impl EdgeModelData {
         let source_width    = default();
         let target_position = default();
         let target_attached = Rc::new(Cell::new(false));
+        let hover_position  = default();
 
         Self {display_object,logger,frp,front,back,source_width,source_height,target_position,
-              target_attached}
+              target_attached,hover_position}
+    }
+
+    fn is_in_upper_half(&self, point:Vector2<f32>) -> bool {
+        let start = nalgebra::Point2::from(self.display_object.position().xy());
+        let end   = nalgebra::Point2::from(self.target_position.get());
+        let point = nalgebra::Point2::from(point);
+        let mid_y = (start.y + end.y) / 2.0;
+        point.y > mid_y
+    }
+
+    /// Returns whether the given hover position belongs to the Input or Output part of the
```

Comment:
Input and Output are written using big letters, so I assume they should be connected to some structs in the code. If so, use backticks to indicate it.